lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -446,23 +413,8 @@ sub get_test_results {
 
     my %locales = $self->{config}->LANGUAGE_locale;
 
-    my $locale;
-    if ( length $language == 2 ) {
-        if ( !exists $locales{$language} ) {
-            die "Undefined language string: '$language'\n";
-        }
-        elsif ( scalar keys %{ $locales{$language} } > 1 ) {
-            die "Language string not unique: '$language'\n";
-        }
-        ( $locale ) = keys %{ $locales{$language} };
-    }
-    else {
-        if ( !exists $locales{substr $language, 0, 2}{$language} ) {
-            die "Undefined language string: '$language'\n";
-        }
-        $locale = $language;
-    }
-    $locale .= '.UTF-8';
+    # Already validated by json_validate
+    my ($locale, undef) = $self->_get_locale($params);
```

Comment:
Have you considered letting the validation layer propagate the normalized values instead of only performing the validation and deferring the normalization until later? In my experience that's a more robust design - you avoid the risk of forgetting to normalize.